dascore/io/rsf/core.py

```diff
@@ -0,0 +1,100 @@
+"""IO module for reading and writing RSF file format support of MADAGASCAR."""
+# from __future__ import annotations
+
+# import segyio
+import datetime as dt
+from pathlib import Path
+
+import numpy as np
+
+import dascore as dc
+from dascore.io.core import FiberIO
+from dascore.utils.time import to_float
+
+RSFKEYS_WRITE = ("in", "esize", "data_format")
+
+
+class RSFV1(FiberIO):
+    """An IO class supporting the RSF format of MADAGASCAR."""
+
+    name = "rsf"
+    preferred_extensions = "rsf"
+    # also specify a version so when version 2 is released you can
+    # just make another class in the same module named rsfV2.
+    version = "1"
+
+    ########################
+    # Add a way to read in rsf files.
+    # May need a way to get absolute time stamp,
+    #   like hide it in header file when writing out an RSF file.
+    ########################
+
+    def write(self, spool, path, data_path=None, **kwargs):
+        """
+        Write a patch to RSF format.
+
+        if no data_path is NOT specified, the header and binary will be
+        packed together
+        data_path needs to be bindata_file.rsf or /location/of/bindata_file.rsf
+        (NO '@')
+
+        path needs to be hdr_file.rsf or /location/of/hdr_file.rsf
+
+        spool needs to have a single patch in it
+        """
+        assert len(spool) == 1
+        patch = spool[0]
+        axis_lengs = patch.shape
+        axis_origs = [to_float(patch.get_coord(x).start) for x in patch.dims]
+        axis_steps = [to_float(patch.get_coord(x).step) for x in patch.dims]
+        axis_names = patch.dims
+        axis_units = [patch.get_coord(x).units.units for x in patch.dims]
+
+        ## we are casting to float32 to ensure that esize=4 for m8r
+        data = patch.data.astype(np.float32)
+        dtype = np.dtype(data.dtype)
+        file_esize = dtype.itemsize
+
+        if np.issubdtype(dtype, np.integer):
+            file_formt = 'data_format="native_int"'
+        elif np.issubdtype(dtype, np.floating):
+            file_formt = 'data_format="native_float"'
+        else:
+            raise ValueError("Data format is not integer or floating.")
```

Comment:
Since we now explicitly convert to float32 dtype I think these lines can just be `file_format = "native_float"` right?